### PR TITLE
[BUG] fix pytorch predict issue and OneHotEncoder compatibility

### LINKED ISSUE #1162: [CI] PyTorch error
In [this run](https://github.com/fairlearn/fairlearn/actions/runs/3573431020/jobs/6007487149) the PyTorch tests fail due to a `RuntimeError`.

```
>       return torch._C._nn.binary_cross_entropy(input, target, weight, reduction_enum)
E       RuntimeError: all elements of input should be between 0 and 1
```

### PR DESCRIPTION
**Pull request by MiroDudik**
## Description
* fix bad predict type in pytorch tests
* comment out bad predict type handling from _lagrangian
* fix OneHotEncoder warning in scikit-learn 1.2
* separate out scikit-learn 1.2 compatibility kwargs into its own module
* increased the batch size in the test of adversarial mitigator to stabilize the learning; closes #1162
* fix a bug in fetch_credit_card

## Tests

- [x] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted

## Documentation

- [x] no documentation changes needed
- [ ] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

File: test_othermlpackages/package_test_common.py [hunk 1]
```diff
@@ -152,6 +155,10 @@ def preprocess(X):
     X = preprocess(X)
     y = preprocess(y)
     sensitive_feature = preprocess(sensitive_feature)
+    # DEBUG
```
Comment (romanlutz) [inline test_othermlpackages/package_test_common.py:158]: Remove?

**Review by romanlutz — APPROVED** [review]

File: test_othermlpackages/package_test_common.py (hunk 1 above)
Comment (MiroDudik) [reply test_othermlpackages/package_test_common.py:158]: yep... once the rest is done... i'm going to mark this as WIP. it seems that one of the pytorch checks is still not passing

**Comment by MiroDudik** [comment]
@fairlearn/fairlearn-maintainers : I believe this one is ready to be merged--it should make all of our tests green again. I'm happy to merge this, but I'd like one more sign-off.

File: fairlearn/reductions/_exponentiated_gradient/_lagrangian.py [hunk 2]
```diff
@@ -28,12 +28,7 @@ def __init__(self, classifier):
         self._classifier = classifier
 
     def __call__(self, X):
-        pred = self._classifier.predict(X)
-        # Some estimators return an output of the shape (num_preds, 1) - flatten such
-        # results
-        if getattr(pred, "flatten", None) is not None:
```
Comment (riedgar-ms) [inline fairlearn/reductions/_exponentiated_gradient/_lagrangian.py:34]: Why is this no longer needed?

File: .github/workflows/build-test.yml [hunk 3]
```diff
@@ -92,6 +92,7 @@ jobs:
       - run: |
           source /usr/share/miniconda/etc/profile.d/conda.sh
           conda activate $condaEnv
+          conda list
```
[the remark this answers is not in the document]
Comment (MiroDudik) [reply .github/workflows/build-test.yml:95]: yes, on the channel pip!

File: fairlearn/reductions/_exponentiated_gradient/_lagrangian.py (hunk 2 above)
Comment (MiroDudik) [reply fairlearn/reductions/_exponentiated_gradient/_lagrangian.py:34]: I think this was an artifact of improperly wrapping pytorch classifiers in our tests. scikit-learn output conventions are here:
https://scikit-learn.org/stable/glossary.html#term-predict
The supported shapes are (n_samples,) and (n_samples, n_outputs), but not (n_samples, 1). As far as I can tell, we have never supported (n_samples, n_outputs) convention (meaning one-hot-encoded predictions)--that might be nice to add, but in a separate PR I think.

**Review by riedgar-ms — APPROVED** [review]

**Pull request merged**